### PR TITLE
[refactor] #3053: Fix clippy lints

### PR DESCRIPTION
**Pull request by Arjentix**
### Description of the Change

Fix some clippy and `cargo check` lints

### Issue

- #3053 

### Benefits

No Mac M1 users have no lint errors hell

### Possible Drawbacks

None

### CODE REVIEW

File: data_model/src/transaction.rs [hunk 1]
```diff
@@ -414,6 +416,7 @@ impl Txn for Transaction {
     FfiType,
     IntoSchema,
 )]
+#[allow(clippy::use_debug)]
```
Comment (Erigara) [inline data_model/src/transaction.rs:419]: We have module level `!#[allow(clippy::use_debug)]` do we need this allow for individual struct.

**Review by Erigara — DISMISSED** [review]
LGTM

File: config/src/path.rs [hunk 2]
```diff
@@ -1,87 +1,85 @@
...
-        Ok(Self(Default(path)))
+    /// # Panics
+    ///
+    /// Panics if `path` contains an extension.
+    #[allow(clippy::panic)]
+    pub fn default(path: &'static std::path::Path) -> Self {
+        assert!(
+            path.extension().is_none(),
+            "Default configuration path should have no extension"
+        );
```
Comment (mversic) [inline config/src/path.rs:60]: While I understand that it reduces boilerplate downstream, I'm not really sure I support this.

**Review by mversic — APPROVED** [review]

File: config/src/path.rs (hunk 2 above)
Comment (Arjentix) [reply config/src/path.rs:60]: I did this because it's a programmer error and we want to know about this as far as we can

Comment (mversic) [reply config/src/path.rs:60]: nvm

File: data_model/src/transaction.rs (hunk 1 above)
Comment (Arjentix) [reply data_model/src/transaction.rs:419]: Indeed. Fixed

**Review by Erigara — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Pull request merged**